File: Kattis/Java/Kornislav.java
package kornislav;

import java.util.Arrays;
import java.util.Scanner;

public class Kornislav {

    public static void main(String[] args) {
       Scanner input = new Scanner(System.in);
       int[] numbers = new int[4];
       
       for (int i = 0; i < 4; i++) {
           numbers[i] = input.nextInt();
       }
       Arrays.sort(numbers);
       System.out.println(numbers[0] * numbers[2]);
    }
    
}

File: Kattis/CPlusPlus/Kornislav.cpp
#include <iostream>
#include <algorithm>

int main() {
	int numbers[4];
	
	for (int i = 0; i < 4; i++) {
		std::cin >> numbers[i];
	}

	std::sort(numbers, numbers+4);
	std::cout << numbers[0] * numbers[2] << std::endl;
	return 0;
}